test/test_genty.py

```diff
@@ -96,6 +96,25 @@ def test_decorated(self, sole_arg):
             )(),
         )
 
+    def test_genty_dataprovider_doesnt_need_any_datasets(self):
+        @genty
+        class SomeClass(object):
+            def my_param_factory(self):
+                return 101
+
+            @genty_dataprovider(my_param_factory)
+            def test_decorated(self, sole_arg):
+                return sole_arg
+
+        instance = SomeClass()
+        self.assertEqual(
+            101,
+            getattr(
+                instance,
+                'test_decorated_{0}'.format('my_param_factory'),
```

Comment:
This is weird because:

a) You're formatting a constant into a constant (so the entire expression is also a constant that you could've just put there).

b) You're doing ``getattr(instance, constant)`` when you could just do ``instance.constant``.